demo-java/app/src/gms/java/com/example/placesdemo/AutocompleteAddressActivity.java

```diff
@@ -0,0 +1,259 @@
+            Log.e(TAG, "Can't find style. Error: ", e);
+        }
+        map.moveCamera(CameraUpdateFactory.newLatLngZoom(coordinates, 15f));
+        marker = map.addMarker(new MarkerOptions().position(coordinates));
+    }
+
+    private void updateMap(LatLng latLng) {
+        marker.setPosition(latLng);
+        map.moveCamera(CameraUpdateFactory.newLatLngZoom(latLng, 15f));
+        if (mapPanel.getVisibility() == View.GONE) {
+            mapPanel.setVisibility(View.VISIBLE);
+        }
+    }
+
+    private void fillInAddress(Place place) {
+        AddressComponents components = place.getAddressComponents();
+        StringBuilder address1 = new StringBuilder();
+        StringBuilder postcode = new StringBuilder();
+
+        // Get each component of the address from the place details,
+        // and then fill-in the corresponding field on the form.
+        // Possible AddressComponent types are documented at https://goo.gle/32SJPM1
+        if (components != null) {
+            for(AddressComponent component : components.asList()) {
+                String type = component.getTypes().get(0);
+                switch(type) {
+                    case "street_number": {
+                        address1.insert(0, component.getName());
+                        break;
+                    }
+
+                    case "route": {
+                        address1.append(" ");
+                        address1.append(component.getShortName());
+                        break;
+                    }
+
+                    case "postal_code": {
+                        postcode.insert(0, component.getName());
+                        break;
+                    }
+
+                    case "postal_code_suffix": {
+                        postcode.append("-").append(component.getName());
+                        break;
+                    }
+
+                    case "locality":
+                        cityField.setText(component.getName());
+                        break;
+
+                    case "administrative_area_level_1": {
+                        stateField.setText(component.getShortName());
+                        break;
+                    }
+
+                    case "country":
+                        countryField.setText(component.getName());
+                        break;
+                }
+            }
+        }
+
+        address1Field.setText(address1.toString());
+        postalField.setText(postcode.toString());
+
+        // After filling the form with address components from the Autocomplete
+        // prediction, set cursor focus on the second address line to encourage
+        // entry of sub-premise information such as apartment, unit, or floor number.
+        address2Field.requestFocus();
+
+        // Add a map for visual confirmation of the address
+        coordinates = place.getLatLng();
+
+        // It isn't possible to set a fragment's id programmatically so we set a tag instead and
```

Comment:
Might be good to pull out this portion of the function in a separation function (maybe something like `showMap(place)`?)

Reply:
done!